apps/web/src/components/AssessmentResultsTable/utils.tsx

```diff
@@ -0,0 +1,270 @@
+  AssessmentStep,
+  AssessmentStepType,
+  PoolSkillType,
+} from "@gc-digital-talent/graphql";
+import { getAssessmentDecision } from "@gc-digital-talent/i18n/src/messages/localizedConstants";
+import { commonMessages } from "@gc-digital-talent/i18n";
+import { notEmpty } from "@gc-digital-talent/helpers";
+
+import poolCandidateMessages from "~/messages/poolCandidateMessages";
+
+import {
+  AssessmentStepResult,
+  AssessmentStepResultColumn,
+  AssessmentStepResultColumnProps,
+  ColumnStatus,
+  StatusColor,
+} from "./types";
+import cells from "../Table/cells";
+import Dialog from "../ScreeningDecisions/ScreeningDecisionDialog";
+
+const iconColorMap: Record<StatusColor, Record<string, string>> = {
+  error: {
+    "data-h2-color": "base(error)",
+  },
+  hold: {
+    "data-h2-color": "base(secondary)",
+  },
+  toAssess: {
+    "data-h2-color": "base(quaternary)",
+  },
+  success: {
+    "data-h2-color": "base(success)",
+  },
+  gray: {
+    "data-h2-color": "base(gray)",
+  },
+};
+
+export const columnHeader = (
+  header: string,
+  status: ColumnStatus,
+  type: AssessmentStepType | null,
+  intl: IntlShape,
+) => {
+  const Icon = status.icon;
+  let ariaLabel;
+  switch (status.color) {
+    case "error":
+      ariaLabel =
+        type === AssessmentStepType.ApplicationScreening ||
+        type === AssessmentStepType.ScreeningQuestionsAtApplication
+          ? intl.formatMessage(poolCandidateMessages.screenedOut)
+          : intl.formatMessage(poolCandidateMessages.unsuccessful);
+      break;
+    case "hold":
+      ariaLabel = intl.formatMessage(poolCandidateMessages.holdForAssessment);
+      break;
+    case "toAssess":
+      ariaLabel = intl.formatMessage(poolCandidateMessages.toAssess);
+      break;
+    case "success":
+      ariaLabel =
+        type === AssessmentStepType.ApplicationScreening ||
+        type === AssessmentStepType.ScreeningQuestionsAtApplication
+          ? intl.formatMessage(poolCandidateMessages.screenedIn)
+          : intl.formatMessage(poolCandidateMessages.successful);
+      break;
+    default:
+      ariaLabel = intl.formatMessage(commonMessages.notApplicable);
+      break;
+  }
+
+  return (
+    <p
+      data-h2-display="base(flex)"
+      data-h2-align-content="base(center)"
+      data-h2-gap="base(x.5)"
+    >
+      {Icon && (
+        <Icon
+          role="img"
+          title={ariaLabel}
+          aria-hidden="false"
+          {...iconColorMap[status.color]}
+          data-h2-display="p-tablet(inline-block)"
+          data-h2-vertical-align="base(middle)"
+          data-h2-height="base(auto)"
+          data-h2-flex-shrink="base(0)"
+          data-h2-width="base(x.85)"
+        />
+      )}
+      <span>{header}</span>
+    </p>
+  );
+};
+
+export const columnStatus = (
+  assessmentStep: AssessmentStep,
+  assessmentResults: AssessmentResult[],
+): ColumnStatus => {
+  // Grab all assessment results from assessment step that have an essential pool skill
```

Comment:
Not quite working right yet, not sure what the issue is yet. As well, should it show "Not found" underneath the "Not demonstrated"? What wasn't found?

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/2666ef67-850d-4f09-afbf-600bc9b8b92c)

Reply:
The user level wasn't found. I'm guessing because of the seeding, or the user tied a skill they don't have, to an experience during the application process. I would say this is out of scope for this issue.

Reply:
I am not too sure about that, looking at the issue. It should just be the first line in red

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/5b081464-322c-4c8d-94f4-2fcf9f5533f4)

Reply:
Okay, the second line comes from this field, which is only applicable to SKILL and SUCCESSFUL results. 

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/24488662-2255-4dee-ab05-46ecf70bf122)